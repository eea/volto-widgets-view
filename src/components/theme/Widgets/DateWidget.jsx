import React from 'react';
import cx from 'classnames';
import moment from 'moment';
import config from '@plone/volto/registry';

export const DateWidget = ({ value, children, className }) => {
  moment.locale(config.settings.dateLocale);
  return value ? (
    <span className={cx(className, 'date', 'widget')}>
      {children
        ? children(moment(value).format('ll'))
        : moment(value).format('ll')}
    </span>
  ) : (
    ''
  );
};
