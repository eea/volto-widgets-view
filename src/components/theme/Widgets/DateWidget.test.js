import React from 'react';
import renderer from 'react-test-renderer';
import { DateWidget } from './DateWidget';

describe('DateWidget', () => {
  it('renders an empty date view widget component', () => {
    const component = renderer.create(<DateWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a date view widget component', () => {
    const component = renderer.create(
      <DateWidget className="metadata">2020-08-18</DateWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
