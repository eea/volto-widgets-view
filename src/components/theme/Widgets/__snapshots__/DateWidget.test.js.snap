// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateWidget renders a date view widget component 1`] = `
<span
  className="metadata date widget"
>
  Aug 18, 2020
</span>
`;

exports[`DateWidget renders an empty date view widget component 1`] = `
<span
  className="date widget"
>
  
</span>
`;
